### PR TITLE
[New] `no-namespace`: Add `ignore` option

### LINKED ISSUE #1903: [no-namespace] allow namespace imports for specific modules
I want to disallow using namespaces as a rule, but some third-party libraries don't use default exports. The specific one that we use quite a bit in our application is [yup](https://github.com/jquense/yup). Even if we use TypeScript's `allowSyntheticDefaultImports` rule, we still have to import the library using `import * as yup from 'yup';`.

I'd like to be able to do something like the following:
```
rules: {
    // ...
    'import/no-namespace': ['warn', {allow: ['yup', 'some-lib', '@scoped/lib']}],
    // ...
},
```

With the above config, the following would be considered correct:
```
import * as yup from 'yup';
import * as validation from 'yup';
import * as lib from 'some-lib';
```

And the following would be considered incorrect:
```
import * as nope from 'not-allowed';

// even though this contains the string "some-lib", it should not be valid; I think the allow list should be exact matches only
import * as stillNope from '@scoped/some-lib';
```

I came across pull request #1679 from earlier this year, but it appears it was closed without being merged.

### PR DESCRIPTION
**Pull request by aberezkin**
Closes #1916
Closes #1903

Hi, we want to use the rule `no-namespace` in our project but we have a `i18n` codegen tool that forces us to use wildcard imports for certain files that follow some pattern (`\S+\.i18n`, to be precise). Also I've seen that there's people with similar issue.

I thought that we can implement something similar to `argsIgnorePattern/varsIgnorePattern` in [core eslint rules](https://github.com/eslint/eslint/blob/master/lib/rules/no-unused-vars.js#L107). 

I made a test for that case and updated the docs as well. Please let me know if there's something else I should do to get this merged.

### CODE REVIEW

**Comment by aberezkin** [comment]
Hi @ljharb! Can you look at this PR? It's quite similar to #1916. But add a little more versatile option.

**Comment by gairon** [comment]
Hi @aberezkin, any updates on that ? Have the same problem with this rule.

**Comment by aberezkin** [comment]
Hi @gairon. I would be happy to provide some update but I'm can't merge it myself since I'm not a maintainer 😄

**Comment by gairon** [comment]
Who else can help with verifying and pushing a really cool feature?
@ljharb @benmosher ?

File: src/rules/no-namespace.js
Comment (ljharb) [inline src/rules/no-namespace.js:34]: eslint rules should never accept regex patterns; this is a common source of ReDOS CVEs. Globs are the only available choice here.

**Review by ljharb — CHANGES_REQUESTED** [review]

File: src/rules/no-namespace.js
Comment (aberezkin) [reply src/rules/no-namespace.js]: Hi! I rewrote it to use minimatch instead of regex.

File: docs/rules/no-namespace.md
Comment (ljharb) [inline docs/rules/no-namespace.md:12]: I'd probably just name this "ignore", and let it be an array of glob strings (the description needs updating)

File: docs/rules/no-namespace.md
Comment (ljharb) [inline docs/rules/no-namespace.md:25]: the example also needs updating

File: src/rules/no-namespace.js
Comment (ljharb) [inline src/rules/no-namespace.js:28]:
```suggestion
          type: 'array',
          items: {
            type: 'string',
          },
          uniqueItems: true,
```

**Comment by aberezkin** [comment]
@ljharb Fixed! :)

**Review by ljharb — APPROVED** [review]

**Comment by gairon** [comment]
Hi @aberezkin,
thanks for new option.
Found a typo in docs:
`/* eslint import/no-namespace: ["error", {ignore: ['*.ext']] */`
there missed `}` before last `]`.

**Comment by ljharb** [comment]
@gairon thanks, fixed in 4f0f560

**Pull request merged**
